sktime/forecasting/compose/_multiplexer.py

```diff
@@ -0,0 +1,188 @@
+
+
+class MultiplexForecaster(
+    _OptionalForecastingHorizonMixin, _SktimeForecaster, _HeterogenousMetaEstimator
+):
+    def __init__(
+        self,
+        forecasters: list,
+        selected_forecaster=None,
+    ):
+
+        """
+        MultiplexForecaster facilitates a framework for performing
+        model selection process over different model classes.
+        It should be used in conjunction with ForecastingGridSearchCV
+        to get full utilization.
+
+        Single use of MultiplexForecaster with forecasters
+        and selected_forecaster parameter specified,
+        works just like the selected component.
+        It does not provide any further use in that case.
+
+        When used with ForecastingGridSearchCV, MultiplexForecaster
+        provides an ability to compare different model class
+        performances with each other, just like a model tournament.
+        When ForecastingGridSearchCV is fitted with a MultiplexForecaster,
+        returned value for the selected_forecaster argument of best_params_
+        attribute of ForecastingGridSearchCV, gives the best
+        performing model class among given models provided in forecasters.
+
+        Parameters
+        ----------
+        forecasters : list
+            List of (forecaster names, forecaster objects)
+            MultiplexForecaster switches between these forecasters
+            objects when used with ForecastingGridSearchCV to
+            find the optimal model
+        selected_forecaster: str
+            An argument to make a selection among forecasters.
+            MultiplexForecaster uses selected_forecaster
+            to choose which component to fit.
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that MultiplexForecaster will currently
+            forecast with.
+        _forecaster_fit_params: dict
+            Fit params for the forecaster that is
+            going to be used by the forecaster in the
+            _forecaster
+        """
+
+        self.forecasters = forecasters
+        self.selected_forecaster = selected_forecaster
+        self._check_components()
+
+        super(MultiplexForecaster, self).__init__()
+
+    def _check_components(self):
+        if not isinstance(self.forecasters, list):
+            raise Exception(
+                "Please provide a list " "for forecasters composed of tuples"
+            )
+
+        for component in self.forecasters:
+            name, forecaster = component
+            if not isinstance(forecaster, _SktimeForecaster):
+                raise Exception(
+                    "Each component has to be an "
+                    "sktime forecaster object. "
+                    "Please check {} with name:{}".format(forecaster, name)
+                )
+
+    def _check_fit_params(self, fit_params):
```

Comment:
I think we need to create our own. Multiplexer switches btw diferent forecasters to find the best performing one. It may be that in this model selection process, user wants to run the process while taking into account different fit_params of different estimators. In scikit_learn, the fit_params argument changes only a param of a single estimator

Reply:
@koralturkk yes you're right!